acceptance/tests/ticket_1429_facter_with_strict_flag.rb

```diff
@@ -0,0 +1,11 @@
+test_name 'ticket FACT-1429  facter should return exit code 1 for querying non-existing fact when --strict is on'
+
+agents.each do |host|
+  step 'facter should return exit code 0 for querying non-existing-fact without --strict flag'
+  on(host, facter('-p non-existing-fact'))
+
+  step 'facter should return exit code 1 for querying non-existing-fact with --strict flag'
+  on(host, facter('-p non-existing-fact --strict'), :acceptable_exit_codes => 1) do
+    assert_match(/ERROR puppetlabs.facter - fact "non-existing-fact" does not exist/, stderr.chomp, 'Unexpected error was detected!')
```

Comment:
@er0ck, I decided to go back to assert_match instead because result.exit_code is a number (either 0 or 1 in this case), so it is very likely to pass the test most of the time regardless of result.exit_code is 0 or 1,  because the output has date/time which usually has both 0 and 1 in it.